src/direct/lwt_direct.ml

```diff
@@ -0,0 +1,117 @@
+module ED = Effect.Deep
+
+module Storage = struct
+  module Lwt_storage=  Lwt.Private.Sequence_associated_storage
+  type 'a key = 'a Lwt.key
+  let new_key = Lwt.new_key
+  let get = Lwt.get
+  let set k v = Lwt_storage.(current_storage := modify_storage k (Some v) !current_storage)
+  let remove k = Lwt_storage.(current_storage := modify_storage k None !current_storage)
+
+  let reset_to_empty () = Lwt_storage.(current_storage := empty_storage)
+  let save_current () = !Lwt_storage.current_storage
+  let restore_current saved = Lwt_storage.current_storage := saved
+end
+
+type _ Effect.t +=
+  | Await : 'a Lwt.t -> 'a Effect.t
+  | Yield : unit Effect.t
+
+(** Queue of microtasks that are ready *)
+let tasks : (unit -> unit) Queue.t = Queue.create ()
+
+let[@inline] push_task f : unit = Queue.push f tasks
+
+let default_on_uncaught_exn exn bt =
+  Printf.eprintf "lwt_task: uncaught task exception:\n%s\n%s\n%!"
+    (Printexc.to_string exn)
+    (Printexc.raw_backtrace_to_string bt)
+
+let run_all_tasks () : unit =
+  let n_processed = ref 0 in
+  let max_number_of_steps = min 10_000 (2 * Queue.length tasks) in
```

Comment:
I think the literal 10_000 is ok for the alpha release. But we'll want to think more about what's the right value at some point and document it. In the meantime we should probably leave a placeholder for a comment and all that jazz just in preparation.

```suggestion
let absolute_max_number_of_steps =
  (* *)
  10_000

let run_all_tasks () : unit =
  let n_processed = ref 0 in
  let max_number_of_steps = min absolute_max_number_of_steps (2 * Queue.length tasks) in
```